File: Microsoft.CodeAnalysis.VisualBasic/Microsoft.CodeAnalysis.VisualBasic.Symbols/PreprocessingSymbol.cs
using System;
using System.Collections.Immutable;

namespace Microsoft.CodeAnalysis.VisualBasic.Symbols
{
	internal sealed class PreprocessingSymbol : Symbol, IPreprocessingSymbol
	{
		private readonly string _name;

		public override string Name => _name;

		public override ImmutableArray<Location> Locations => ImmutableArray<Location>.Empty;

		public override ImmutableArray<SyntaxReference> DeclaringSyntaxReferences => Symbol.GetDeclaringSyntaxReferenceHelper<VisualBasicSyntaxNode>(Locations);

		public override SymbolKind Kind => SymbolKind.Preprocessing;

		public override Symbol ContainingSymbol => null;

		public override Accessibility DeclaredAccessibility => Accessibility.NotApplicable;

		public override bool IsMustOverride => false;

		public override bool IsNotOverridable => false;

		public override bool IsOverridable => false;

		public override bool IsOverrides => false;

		public override bool IsShared => false;

		internal override ObsoleteAttributeData ObsoleteAttributeData => null;

		internal PreprocessingSymbol(string name)
		{
			_name = name;
		}

		public override bool Equals(object obj)
		{
			if (obj == this)
			{
				return true;
			}
			if (obj == null)
			{
				return false;
			}
			return obj is PreprocessingSymbol preprocessingSymbol && CaseInsensitiveComparison.Equals(Name, preprocessingSymbol.Name);
		}

		public override int GetHashCode()
		{
			return Name.GetHashCode();
		}

		public override void Accept(SymbolVisitor visitor)
		{
			throw new NotSupportedException();
		}

		public override void Accept(VisualBasicSymbolVisitor visitor)
		{
			throw new NotSupportedException();
		}

		public override TResult Accept<TResult>(SymbolVisitor<TResult> visitor)
		{
			throw new NotSupportedException();
		}

		public override TResult Accept<TResult>(VisualBasicSymbolVisitor<TResult> visitor)
		{
			throw new NotSupportedException();
		}

		internal override TResult Accept<TArgument, TResult>(VisualBasicSymbolVisitor<TArgument, TResult> visitor, TArgument arg)
		{
			throw new NotSupportedException();
		}
	}
}

File: Microsoft.CodeAnalysis.VisualBasic/Microsoft.CodeAnalysis/VisualBasic/Symbols/PreprocessingSymbol.vb
Imports Microsoft.CodeAnalysis
Imports Microsoft.CodeAnalysis.VisualBasic
Imports System
Imports System.Collections.Immutable

Namespace Microsoft.CodeAnalysis.VisualBasic.Symbols
	Friend NotInheritable Class PreprocessingSymbol
		Inherits Symbol
		Implements IPreprocessingSymbol
		Private ReadOnly _name As String

		Public Overrides ReadOnly Property ContainingSymbol As Symbol
			Get
				Return Nothing
			End Get
		End Property

		Public Overrides ReadOnly Property DeclaredAccessibility As Accessibility
			Get
				Return Accessibility.NotApplicable
			End Get
		End Property

		Public Overrides ReadOnly Property DeclaringSyntaxReferences As ImmutableArray(Of SyntaxReference)
			Get
				Return Symbol.GetDeclaringSyntaxReferenceHelper(Of VisualBasicSyntaxNode)(Me.Locations)
			End Get
		End Property

		Public Overrides ReadOnly Property IsMustOverride As Boolean
			Get
				Return False
			End Get
		End Property

		Public Overrides ReadOnly Property IsNotOverridable As Boolean
			Get
				Return False
			End Get
		End Property

		Public Overrides ReadOnly Property IsOverridable As Boolean
			Get
				Return False
			End Get
		End Property

		Public Overrides ReadOnly Property IsOverrides As Boolean
			Get
				Return False
			End Get
		End Property

		Public Overrides ReadOnly Property IsShared As Boolean
			Get
				Return False
			End Get
		End Property

		Public Overrides ReadOnly Property Kind As SymbolKind
			Get
				Return SymbolKind.Preprocessing
			End Get
		End Property

		Public Overrides ReadOnly Property Locations As ImmutableArray(Of Location)
			Get
				Return ImmutableArray(Of Location).Empty
			End Get
		End Property

		Public Overrides ReadOnly Property Name As String
			Get
				Return Me._name
			End Get
		End Property

		Friend Overrides ReadOnly Property ObsoleteAttributeData As Microsoft.CodeAnalysis.ObsoleteAttributeData
			Get
				Return Nothing
			End Get
		End Property

		Friend Sub New(ByVal name As String)
			MyBase.New()
			Me._name = name
		End Sub

		Public Overrides Sub Accept(ByVal visitor As SymbolVisitor)
			Throw New NotSupportedException()
		End Sub

		Public Overrides Sub Accept(ByVal visitor As VisualBasicSymbolVisitor)
			Throw New NotSupportedException()
		End Sub

		Public Overrides Function Accept(Of TResult)(ByVal visitor As SymbolVisitor(Of TResult)) As TResult
			Throw New NotSupportedException()
		End Function

		Public Overrides Function Accept(Of TResult)(ByVal visitor As VisualBasicSymbolVisitor(Of TResult)) As TResult
			Throw New NotSupportedException()
		End Function

		Friend Overrides Function Accept(Of TArgument, TResult)(ByVal visitor As VisualBasicSymbolVisitor(Of TArgument, TResult), ByVal arg As TArgument) As TResult
			Throw New NotSupportedException()
		End Function

		Public Overrides Function Equals(ByVal obj As Object) As Boolean
			Dim flag As Boolean
			If (obj = Me) Then
				flag = True
			ElseIf (obj IsNot Nothing) Then
				Dim preprocessingSymbol As Microsoft.CodeAnalysis.VisualBasic.Symbols.PreprocessingSymbol = TryCast(obj, Microsoft.CodeAnalysis.VisualBasic.Symbols.PreprocessingSymbol)
				flag = If(preprocessingSymbol Is Nothing, False, CaseInsensitiveComparison.Equals(Me.Name, preprocessingSymbol.Name))
			Else
				flag = False
			End If
			Return flag
		End Function

		Public Overrides Function GetHashCode() As Integer
			Return Me.Name.GetHashCode()
		End Function
	End Class
End Namespace